integration_tests/src/main/python/parquet_testing_test.py

```diff
@@ -72,6 +75,53 @@
     _error_files["lz4_raw_compressed.parquet"] = "Exception"
     _error_files["lz4_raw_compressed_larger.parquet"] = "Exception"
 
+def hdfs_glob(path, pattern):
+    """
+    Finds hdfs files by checking the input path with glob pattern
+
+    :param path: hdfs path to check 
+    :type path: pathlib.Path 
+    :return: generator of matched files
+    """
+    path_str = path.as_posix()
+    cmd = ['hadoop', 'fs', '-ls', '-R', path_str]
+
+    process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    stdout, stderr = process.communicate()
+    if process.returncode != 0:
+        warnings.warn('Failed to list files from hdfs: {}. Error: {}'.format(path_str, stderr))
+        return iter(())
+    
+    lines = stdout.strip().split('\n')
+    paths = [line.split()[-1] for line in lines]
+    compiled_pattern = re.compile(fnmatch.translate(pattern), re.IGNORECASE)
+
+    for p in paths:
+        p = Path(p)
+        relative_path = p.relative_to(path_str).as_posix()
+        if compiled_pattern.match(relative_path):
+            yield p
+
+def glob(path, pattern):
+    """
+    Finds files by checking the input path with glob pattern.
+    Support local file system and hdfs
+
+    :param path: input path to check 
+    :type path: pathlib.Path 
+    :return: generator of matched files
+    """
+    path_str = path.as_posix()
+    if not path_str.startswith('hdfs:'):
+        return path.glob(pattern)
+
+    try:
+       return hdfs_glob(path, pattern)
+    except Exception as e:
+        warnings.warn('Failed to run glob path from hdfs: {}. Error: {}'.format(path_str, str(e)))
```

Comment:
I don't think we want this exception handling here.  If we are told to look in an hdfs:// path and we cannot even run `hadoop`, IMO the tests should fail rather than ignore that basic failure.  Note there's no exception handling on the local path, because we want the tests to fail if we're getting I/O errors or other oddities during this processing.

Arguably we don't even want to suppress the failure to list any files in the HDFS paths.  The only reason we put in the listing suppression logic originally is because some users may not bother to do a submodule init/update on the parquet_testing submodule and not all CI pipelines were ready to do the submodule init/update when this was first added.  If we're running with HDFS paths, then that means we're using the integration tests tgz unpacked into an HDFS directory.  In that case we fully expect everything the tests need should be there, and thus we would not expect the parquet_testing files to be missing in the HDFS input files scenario.  Something is wrong with the test setup if those files are missing in HDFS paths.

Reply:
Removed the exception handling here to expose the failure. Thanks for the explanation!